### DISCUSSION TITLE
Inconsistent object selection

### DISCUSSION THREAD

**Discussion by ll931217** [fabricjs/fabric.js#6896] [Q&A] [answered]
Hello,

This issue could be my own error, I have an object that is at point A, I changed it's position by using:

```javascript
canvas.forEachObject(function(object) {
  if (object.id === obj.id) {
    object.setOptions({ top: obj[key] })
  }
})

// This code shows that the top value has changed correctly
canvas.getObjects().find(obj => obj.id === 'C001')
```

What I see on my canvas is this:
* Point in original position. 
![219969](https://user-images.githubusercontent.com/12750413/109124611-15343c80-7786-11eb-855d-a80e4ef6c430.jpg)

* Point moved, but I am unable to select it. 
![219970](https://user-images.githubusercontent.com/12750413/109124620-17969680-7786-11eb-9a56-4233c3412c32.jpg)

* I am able to select the point from the original position.
![219971](https://user-images.githubusercontent.com/12750413/109124633-19f8f080-7786-11eb-9c9d-30b5b2c90491.jpg)

I thought it was a caching issue because I saw that it has a property `ownCaching` but after adding `objectCaching: false`, the problem still persists. 

What fixed it is by calling `canvas.renderAll()` again, which happens after I finish changing the `top` value, I deselect the object, select the object and deselect it again. This process is very tedious when I have many objects that needs to change their `top` value.

Unfortunately, I am unable to reproduce this, which could mean my code has an issue. I am posting this to see if anyone can guide me to where my issue could lie.

# Version
4.3.1

**Comment by asturur** [comment, marked as answer]
hi, setOptions is an internal method. Pleas use the simple set ( it takes an object or a single key, value ).

What you really need to fix is described in this page here
http://fabricjs.com/fabric-gotchas

**Reply by ll931217** [reply]
Hello @asturur,

Thank you for the quick response

I changed my code as you suggested, however that didn't seem to do the trick. I am not sure if I this helps but the code mentioned above is used inside a jQuery `keyup` event callback.

```javascript
$('#someinput').keyup(function() {
  canvas.forEachObject(function(object) {
    if (object.id === obj.id) {
      // object.setOptions({ left: obj[key] })
      // object.set('left', obj[key])
      object.left = obj[key]
      object.set('dirty', true)
    }
  })
})
```

**Reply by asturur** [reply]
you did not read all the page i linked. pleas read it all is inportant to me to understand if that clarifies

**Reply by ll931217** [reply]
Ah! Sorry, adding `object.setCoords()` fixed the issue for me. Thank you for your time.
